rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -1519,7 +1622,16 @@ void PlayerImpl::publish_clock_update(const rclcpp::Time & time)
 
 const rosbag2_storage::StorageOptions & PlayerImpl::get_storage_options()
 {
-  return storage_options_;
+  return input_bags_[0].second;
+}
+
+std::vector<rosbag2_storage::StorageOptions> PlayerImpl::get_all_storage_options()
+{
+  std::vector<rosbag2_storage::StorageOptions> storage_options{};
+  for (const auto & [_, options] : input_bags_) {
```

Comment:
Preallocate memory for the `storage_options`
```suggestion
  std::vector<rosbag2_storage::StorageOptions> storage_options{};
  storage_options.reserve(input_bags_.size());
  for (const auto & [_, options] : input_bags_) {
```

Reply:
changed in 82814f011d9f938889bc50d4d9060b95b1de0689